app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -18,6 +18,14 @@ def idme
       return redirect_to spouse_identity_questions_path(missing_spouse: "true")
     end
 
+    if is_returning_consenting_user || is_returning_nonconsenting_user || is_new_spouse
```

Comment:
thoughts on simplifying this to:

`if session[:intake_id] && (is_returning_user || is_new_spouse)`

?

Also- as an idea- you could create a new variable for this situation:

```ruby
# up above:
should_delete_duplicate_intake = session[:intake_id] && (is_returning_user || is_new_spouse)

# right here:
if should_delete_duplicate_intake
  # ...
end
```